webapp/utils/cipher.py

```diff
@@ -0,0 +1,26 @@
+import base64
+import hashlib
+
+from cryptography.fernet import Fernet, InvalidToken
+
+
+class Cipher:
+    def __init__(self, secret):
+        self.cipher_suite = Fernet(
+            base64.urlsafe_b64encode(
+                hashlib.sha256(secret.encode("utf-8")).digest()
```

Comment:
Why do we need to hash this?

Reply:
Because the secret key length must be 32 characters
So I hash to make sure it respects the length

Reply:
Yeah, but we can just make sure the secret is 32 bytes long can't we? Here you're adding processing overhead to every request. I agree this is hopefully insignificant but it's still not good to do if it can be avoided.

Secrets are always something that needs to be very specific. It's not like people can just expect to choose a random string. So expecting the secret to be stored at 32 bytes long doesn't seem to be particularly unreasonable to me...

Reply:
32 bytes long secret is a requirement of the cryptography library, having a secret that doen't match the requirement throw an errror. This is calculated only once at the moment we instanciate the class.

I don't mind removing this line, the problem is that we will have to update all the applications encrypted token with a new one which @tbille spent a few hours on last time!

Reply:
*sigh* fine. It's merged now anyway.